### PR TITLE
Ahunt/32 external

### PR DESCRIPTION
**Pull request by ahunt**
Allow opening external links

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/utils/IntentUtils.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        builder.setNegativeButton(R.string.external_app_prompt_cancel, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(final DialogInterface dialog, final int which) {
+                dialog.dismiss();
+            }
+        });
+
+        // TODO: remove this, or enable it - depending on how we decide to handle the multiple-app/>1
+        // case in future.
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/focus/utils/IntentUtils.java]: Should we file an issue for that?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -93,4 +93,14 @@
     <string name="firstrun_description">Automatically block online trackers while you browse. Then tap to erase visited pages, searches, cookies and passwords from your device.</string>
     <string name="firstrun_button">Ok, got it!</string>
     <string name="firstrun_footer">To be clear, sites and service you use can still know where you\'ve been.</string>
+
+    <!-- This string is shown when the user has clicked a link that needs to be opened in a different app.
+         Argument 2 is the external app name, argument 1 is our name (i.e. Firefox Focus, or Firefox Klar). -->
+    <string name="external_app_prompt">This link will open in %2$s. Are you sure you want to exit %1$s?</string>
```
Comment (pocmo) [inline app/src/main/res/values/strings.xml]: It's a bit confusing that in the original string argument 2 comes before argument 1. But with the comment it shouldn't be a problem.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -93,4 +93,14 @@
...
+    <string name="external_app_prompt_no_app_title">No app to handle link</string>
+    <string name="external_app_prompt_no_app">None of the apps on your device are able to open this link. Do you want to exit %1$s and open %2$s to find a suitable app?</string>
+    <string name="external_app_prompt_ok">Ok</string>
+    <string name="external_app_prompt_cancel">Cancel</string>
```
Comment (pocmo) [inline app/src/main/res/values/strings.xml]: I also added a 'Cancel' string for a dialog in one of my patches that I'm currently landing. We should make sure to unify them and just have one string before exporting them again.

File: app/src/main/java/org/mozilla/focus/utils/IntentUtils.java (hunk 1 above)
Comment (ahunt) [inline app/src/main/java/org/mozilla/focus/utils/IntentUtils.java]: Filed:
https://github.com/mozilla-mobile/focus-android/issues/223

**Pull request merged**